src/Features/Core/Portable/LanguageServices/SymbolDisplayService/AbstractSymbolDisplayService.AbstractSymbolDescriptionBuilder.cs

```diff
@@ -503,10 +503,10 @@ private void AddDescriptionForNamedType(INamedTypeSymbol symbol)
 
         private static bool TypeArgumentsAndParametersAreSame(INamedTypeSymbol symbol)
         {
-            var typeArguments = symbol.GetAllTypeArguments().ToList();
-            var typeParameters = symbol.GetAllTypeParameters().ToList();
+            var typeArguments = symbol.GetAllTypeArguments();
+            var typeParameters = symbol.GetAllTypeParameters();
 
-            for (var i = 0; i < typeArguments.Count; i++)
+            for (var i = 0; i < typeArguments.Length; i++)
```

Comment:
>for (var i = 0; i < typeArguments.Length; i++) [](http://example.com/codeflow?start=12&length=46)

nit: preference for me is to have the return false inside the condition. Always looks odd to me when a loop has the initial appearance of not looping more than once.